lib/api/completeMultipartUpload.js

```diff
@@ -119,14 +134,22 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                 // Required permissions for this action
                 // at the destinationBucket level are same as objectPut
                 requestType: 'objectPut',
+                versionId,
             };
             metadataValidateBucketAndObj(metadataValParams, log, next);
         },
         function validateMultipart(destBucket, objMD, next) {
             if (objMD) {
                 oldByteLength = objMD['content-length'];
             }
-            services.metadataValidateMultipart(metadataValParams,
+
+            if (isPutVersion && !objMD) {
+                const err = putVersionId ? errors.NoSuchVersion : errors.NoSuchKey;
+                log.trace('error no object metadata found', { method: 'completeMultipartUpload', putVersionId });
+                return callback(err);
```

Comment:
Any reason we do not have any monitoring in case of error in this API?

Reply:
Good catch, adding a JIRA ticket https://scality.atlassian.net/browse/CLDSRV-206 since it is out of the scope of this PR.